### PR TITLE
Refactor branch status modification with new generic design

### PR DESCRIPTION
**Pull request by dbraquart**

### CODE REVIEW

File: src/main/java/org/gridsuite/modification/server/modifications/BranchStatusModification.java [hunk 1]
```diff
@@ -0,0 +1,148 @@
...
+            .withKey("lockout" + branchType + "Applied")
+            .withDefaultMessage(branchType + " ${id} (id) : lockout applied")
+            .withValue("id", branch.getId())
+            .withSeverity(TypedValue.INFO_SEVERITY)
+            .build());
+    }
+
+    private void applyTripBranch(Reporter subReporter, Branch<?> branch, String branchType, Network network) {
+        var trip = new BranchTripping(branch.getId());
+        var switchToDisconnect = new HashSet<Switch>();
```
Comment (laurgarn) [inline src/main/java/org/gridsuite/modification/server/modifications/BranchStatusModification.java:82]: A mark  of plural (-> `switches` I guess) would be welcome

Comment (dbraquart) [reply src/main/java/org/gridsuite/modification/server/modifications/BranchStatusModification.java:82]: its just a move of existing code, but you're right. I will change

File: src/main/java/org/gridsuite/modification/server/NetworkModificationController.java [hunk 2]
```diff
@@ -109,38 +109,14 @@ public ResponseEntity<List<UUID>> getModificationGroups() {
...
-            case LINES_ATTACH_TO_SPLIT_LINES:
-            case GENERATOR_CREATION:
-            case GENERATOR_MODIFICATION:
-            case SUBSTATION_CREATION:
-            case TWO_WINDINGS_TRANSFORMER_CREATION:
-                return ResponseEntity.ok().body(networkModificationService.createModification(networkUuid, variantId, groupUuid, reportUuid, reporterId, modificationInfos));
-            default:
-                return ResponseEntity.ok().body(networkModificationService.createNetworkModification(networkUuid, variantId, groupUuid, reportUuid, reporterId, modificationInfos));
-        }
+        return ResponseEntity.ok().body(networkModificationService.createModification(networkUuid, variantId, groupUuid, reportUuid, reporterId, modificationInfos));
```
Comment (SlimaneAmar) [inline src/main/java/org/gridsuite/modification/server/NetworkModificationController.java:119]: Rename `createModification` method by `createNetworkModification` in the controller and the service

File: src/main/java/org/gridsuite/modification/server/service/NetworkModificationService.java [hunk 3]
```diff
@@ -540,13 +332,8 @@ private List<ModificationInfos> applyModification(NetworkStoreListener listener,
                 case GENERATOR_MODIFICATION:
                 case SUBSTATION_CREATION:
                 case TWO_WINDINGS_TRANSFORMER_CREATION:
-                    // Generic form
+                case BRANCH_STATUS_MODIFICATION:
```
Comment (SlimaneAmar) [inline src/main/java/org/gridsuite/modification/server/service/NetworkModificationService.java:335]: Remove the switch

Comment (SlimaneAmar) [inline src/main/java/org/gridsuite/modification/server/service/NetworkModificationService.java:335]: Remove the `doAction` method

**Review by SlimaneAmar — CHANGES_REQUESTED** [review]

File: src/main/java/org/gridsuite/modification/server/service/NetworkModificationService.java (hunk 3 above)
Comment (dbraquart) [reply src/main/java/org/gridsuite/modification/server/service/NetworkModificationService.java:335]: ok.
(I was hesitating to start the code cleaning)

**Review by SlimaneAmar — APPROVED** [review]

**Pull request merged**